unit/analyses/cfg_dominators.cpp

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************\
+
+Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <string>
```

Comment:
Testing: how about retaining the existing dominator algorithm, which is simple enough to check by inspection, randomly generating a couple of hundred medium-sized graphs and checking they give identical results?

Reply:
@smowton That's more or less what I did manually when testing this IIRC, but doing this as part of automated testing of course makes a lot of sense